In [1]:
import sklearn as sk
import pandas as pd
import numpy as np

In [2]:
from sklearn.metrics import f1_score
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score
from sklearn.model_selection import train_test_split


In [29]:
result2=pd.read_csv('result_2.csv')
result5=pd.read_csv('result_5.csv')
result10=pd.read_csv('result_1000.csv')
result5_cleaned=pd.read_csv('result5_cleaned.csv')
result1_cleaned=pd.read_csv('result1_cleaned.csv')

In [5]:
result2.head()

true  pred      frac
0   1.0     1  0.707712
1   0.0     0  0.379577
2   0.0     0  0.534948
3   0.0     0  0.416692
4   0.0     0  0.420347

In [8]:
def fun(df_eval):
    cnt=0
    scr=0
    for i in range(df_eval.shape[0]):
        n=i
        if(df_eval["true"].iloc[n]==1):
            cnt+=1
            scr+=df_eval["frac"].iloc[n]
    print("Average = ",scr/cnt)
    print("Accuracy = ",accuracy_score(df_eval['true'], df_eval['pred']))
    print("F1 score = ",f1_score(df_eval['true'], df_eval['pred'], zero_division=1))
    print("Confusion metric ")
    print(confusion_matrix(df_eval['true'], df_eval['pred']))


In [9]:
fun(result5)

Average =  0.6145481270219931
Accuracy =  0.7426666666666667
F1 score =  0.44487056567593475
Confusion metric 
[[5756  243]
 [2073  928]]


In [10]:
fun(result10)

Average =  0.6864851313070981
Accuracy =  0.7505555555555555
F1 score =  0.5901040715720285
Confusion metric 
[[5139  860]
 [1385 1616]]


In [11]:
fun(result2)

Average =  0.6588369196863071
Accuracy =  0.7478888888888889
F1 score =  0.5623915139826422
Confusion metric 
[[5273  726]
 [1543 1458]]


In [12]:
df_eval=pd.read_excel("evaluation.xlsx")
df_eval.head()


text  \
0            the app is crashing when i play a vedio   
1  but i want to connect it to the tv from one de...   
2        very helpful when and home working remotley   
3  this zoom so called and missed call and mobile...   
4                            one of my favorite apps   

                                           reason  label  
0                     app crashes during playback    1.0  
1  want compatibility with more smart televisions    0.0  
2                               good app for work    0.0  
3        receiving incorrect phone number message    0.0  
4                          good for spending time    0.0

In [13]:
df_eval=result2

In [14]:
X_train, X_test, y_train, y_test = train_test_split(np.array(df_eval['frac'].values), np.array(df_eval['true'].values), test_size=0.33, random_state=42,stratify=np.array(df_eval['true'].values))

Testing with Different Metrices and differnt models for the best fit and threshold on the data

In [21]:

def find_thresh(df_eval,scr):
    X_train, X_test, y_train, y_test = train_test_split(np.array(df_eval['frac'].values), np.array(df_eval['true'].values), test_size=0.33, random_state=42,stratify=np.array(df_eval['true'].values))
    resu=[-1,0]
    y_pred_test=[]
    for i in range(20):
        y_pred_train=[]
        vl=i*0.05
        for j in range(y_train.shape[0]):
            y_pred_train.append((1 if X_train[j] >= vl else 0))
        f1=-10
        if scr==1:
            f1=f1_score(list(y_train),y_pred_train)
        else :
            f1=accuracy_score(list(y_train),y_pred_train)
        if(resu[0]<=f1):
            resu=[f1,vl]

    for j in range(y_test.shape[0]):
            y_pred_test.append((1 if X_test[j] >= resu[1] else 0))

    print("F1 Score is = " if scr==1 else "Accuracy Score is = "  ,resu[0]," at thresh = ",resu[1])
    print("Accuracy = ",accuracy_score(list(y_test), y_pred_test))
    print("F1 score = ",f1_score(list(y_test), y_pred_test, zero_division=1))
    print("Confusion metric ")
    print(confusion_matrix(list(y_test), y_pred_test))
    
        

    

In [22]:
find_thresh(result2,1)

F1 Score is =  0.6256935552710201  at thresh =  0.55
Accuracy =  0.7063973063973064
F1 score =  0.6211989574283231
Confusion metric 
[[1383  597]
 [ 275  715]]


In [23]:
find_thresh(result2,0)

Accuracy Score is =  0.7446102819237148  at thresh =  0.7000000000000001
Accuracy =  0.7548821548821549
F1 score =  0.5640718562874251
Confusion metric 
[[1771  209]
 [ 519  471]]


In [24]:
find_thresh(result5,1)

F1 Score is =  0.6537911301859801  at thresh =  0.55
Accuracy =  0.763973063973064
F1 score =  0.6624939817043813
Confusion metric 
[[1581  399]
 [ 302  688]]


In [25]:
find_thresh(result5,0)

Accuracy Score is =  0.7704809286898839  at thresh =  0.6000000000000001
Accuracy =  0.7848484848484848
F1 score =  0.6395939086294417
Confusion metric 
[[1764  216]
 [ 423  567]]


In [26]:
find_thresh(result10,1)

F1 Score is =  0.6178416609628109  at thresh =  0.6000000000000001
Accuracy =  0.7333333333333333
F1 score =  0.6323119777158774
Confusion metric 
[[1497  483]
 [ 309  681]]


In [27]:
find_thresh(result10,0)

Accuracy Score is =  0.7487562189054726  at thresh =  0.75
Accuracy =  0.7572390572390573
F1 score =  0.5416401780038144
Confusion metric 
[[1823  157]
 [ 564  426]]


In [30]:
find_thresh(result1_cleaned,0)

Accuracy Score is =  0.7704809286898839  at thresh =  0.6000000000000001
Accuracy =  0.7848484848484848
F1 score =  0.6395939086294417
Confusion metric 
[[1764  216]
 [ 423  567]]


In [31]:
find_thresh(result1_cleaned,1)

F1 Score is =  0.6537911301859801  at thresh =  0.55
Accuracy =  0.763973063973064
F1 score =  0.6624939817043813
Confusion metric 
[[1581  399]
 [ 302  688]]


In [33]:
find_thresh(result5_cleaned,1)

F1 Score is =  0.6537911301859801  at thresh =  0.55
Accuracy =  0.763973063973064
F1 score =  0.6624939817043813
Confusion metric 
[[1581  399]
 [ 302  688]]


In [34]:
find_thresh(result5_cleaned,0)

Accuracy Score is =  0.7704809286898839  at thresh =  0.6000000000000001
Accuracy =  0.7848484848484848
F1 score =  0.6395939086294417
Confusion metric 
[[1764  216]
 [ 423  567]]


Result : 
Model : Pegasus Large 
Epoch : 1
Threshold : .60
Accuracy : 78.5
F1 Score : .64
Changes In Data : Data Cleaned Using Regex